### PR TITLE
ODC-6445: Visualize kamelets sinks

### PR DESCRIPTION
**Pull request by sahil143**
This PR
- visualize Kamelet's sinks and event sink connector
- sidebar for kamelet sinks

[TODO]:

- [ ] fix event sink connector

Screenshots:
Screenshot 2022-01-12 at 9 01 50 PM
Screenshot 2022-01-12 at 9 02 01 PM
Screenshot 2022-01-12 at 9 01 20 PM

Test Setup:

- Install Camel K operator
- Create a Channel
- Create Kamelet Sink using below yaml
```
apiVersion: camel.apache.org/v1alpha1
kind: KameletBinding
metadata:
  name: log-sink-binding
spec:
  source:
    ref:
      kind: Channel
      apiVersion: messaging.knative.dev/v1
      name: channel
  sink:
    ref:
      kind: Kamelet
      apiVersion: camel.apache.org/v1alpha1
      name: log-sink
```

### CODE REVIEW

File: frontend/packages/knative-plugin/src/topology/components/edges/EventSinkLink.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+type EventSinkLinkProps = {
+  element: Edge;
+};
+
+const EventSinkLink: React.FC<EventSinkLinkProps> = ({ element, ...others }) => {
+  const markerPoint = element.getStartPoint();
+
+  return (
+    <BaseEdge className="odc-event-source-link" element={element} {...others}>
```
Comment (karthikjeeyar) [inline frontend/packages/knative-plugin/src/topology/components/edges/EventSinkLink.tsx:15]: nit:
```suggestion
    <BaseEdge className="kn-event-source-link" element={element} {...others}>
```

File: frontend/packages/knative-plugin/src/topology/components/nodes/EventSink.tsx [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        ref={svgAnchorRef}
+        filter={createSvgIdUrl(
+          hover || dragging || contextMenuOpen
+            ? NODE_SHADOW_FILTER_ID_HOVER
+            : NODE_SHADOW_FILTER_ID,
+        )}
+        points={`${width / 2}, ${(height - size) / 2} ${width - (width - size) / 2},${height /
+          2} ${width / 2},${height - (height - size) / 2} ${(width - size) / 2},${height / 2}`}
+      />
+      {donutStatus && <PodSet size={size * 0.75} x={width / 2} y={height / 2} data={donutStatus} />}
```
Comment (karthikjeeyar) [inline frontend/packages/knative-plugin/src/topology/components/nodes/EventSink.tsx:132]: Looks like except the PodSet the most of other elements are same for Eventsource.tsx, if yes, can we abstract out a generic BasePolygonNode and make the code DRY WDYT?

File: frontend/packages/knative-plugin/src/topology/knative-topology-utils.ts [hunk 3]
```diff
@@ -747,6 +749,32 @@ const getEventSourcesData = (sinkUri: string, resources) => {
 };
 
 const getApiGroup = (apiVersion: string) => groupVersionFor(apiVersion)?.group;
+
+export const getEventSinkTopologyEdgeItems = (resource: K8sResourceKind, resources) => {
```
Comment (karthikjeeyar) [inline frontend/packages/knative-plugin/src/topology/knative-topology-utils.ts:753]: Good to have test cases for the functions in this utils file.

**Review by karthikjeeyar — COMMENTED** [review]
Clicking on KB resource, the causes the below error and the page goes unresponsive after a while. 
image

**Comment by serenamarie125** [comment]
@sahil143 this is looking good!  Could you share a mock of what this looks like when the pod is running?

**Comment by nainaz** [comment]
This looks good. Thank you @sahil143 
Same question as Chris. Edge line looks like just hanging in the ether.

**Comment by sahil143** [comment]
@christianvogt @nainaz I'm working on fixing the edge line.

**Comment by sahil143** [comment]
@serenamarie125 added screenshot with pod running

image
Screenshot 2022-01-13 at 10 04 36 PM

**Comment by sahil143** [comment]
@christophd We decided to use `right-from-bracket` for event sources. Here are the designs https://docs.google.com/document/d/1xJ_oYhfFDJO8G9iDaBO8LvfIfp0GIBQWXFnE4M2DjsQ/edit#heading=h.2tz70k6xknyz

cc @serenamarie125

**Comment by invincibleJai** [comment]
Pls refer to https://github.com/openshift/console/pull/10892

/close

**Pull request closed without merging**